gcp/modules/couchdb/main.tf

```diff
@@ -2,10 +2,64 @@ terraform {
   backend "gcs" {}
 }
 
+variable "couchdb_prd_replicas" {
+  default = 3
+}
+
+variable "couchdb_dev_replicas" {
+  default = 2
+}
+
+variable "release_namespace" {
+  default = "gpii"
+}
+
+variable "env" {}
 variable "secrets_dir" {}
 variable "values_dir" {}
-variable "release_namespace" {
-	default = "gpii"
+variable "project_id" {}
+variable "serviceaccount_key" {}
+
+provider "google" {
+  project     = "${var.project_id}"
+  credentials = "${var.serviceaccount_key}"
+}
+
+resource "google_compute_disk" "couchdb" {
+  count = "${var.env == "dev" ? var.couchdb_dev_replicas : var.couchdb_prd_replicas}"
+  name  = "couchdb-${count.index}"
+  type  = "pd-ssd"
+  size  = 10
+  zone  = "us-central1-a"
+}
+
+data "template_file" "couchdb_pvs" {
+  template = "${file("resources/pv.yaml")}"
+  count    = "${var.env == "dev" ? var.couchdb_dev_replicas : var.couchdb_prd_replicas}"
+
+  vars {
+    env   = "${var.env}"
+    index = "${count.index}"
+  }
+}
+
+resource "null_resource" "couchdb_pvs" {
+  depends_on = ["google_compute_disk.couchdb"]
+
+  provisioner "local-exec" {
+    command = <<EOF
+      echo "${join("", data.template_file.couchdb_pvs.*.rendered)}" | kubectl create -f -
+    EOF
+  }
+
+  provisioner "local-exec" {
+    when    = "destroy"
+    command = <<EOF
+      sleep 30 # Since there is no way to establish a proper dependency for module, we need to give couchdb helm release time to uninstall
```

Comment:
This seems kind of brittle. Can we solve this by 1) moving the PV management code to its own module and 2) using terragrunt dependencies so that the PVs and the couchdb chart are installed/uninstalled in the correct order?